marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/WebMvcConfig.java

```diff
@@ -44,4 +46,14 @@ protected Resource getResource(String resourcePath, Resource location)
                 }
             });
     }
+
+    @Override
+    public void configureAsyncSupport(AsyncSupportConfigurer configurer) {
```

Comment:
Is there something under src/main/java that needs this? The only reference I saw to "async" is in EntitySearchControllerTest.

Reply:
We need this since we are using a StreamingResponseBody for asynchronous request processing.
StreamingResponseBody by default uses a SimpleAsyncTaskExecutor which does not reuse threads, as a result, its advisable to provide our own TaskExecutor. 
Also, this allows us to configure the Default Timeout which at this point is 10 minutes.

Reply:
I think this method needs to make that clear. Spring is great at making it easy to setup useful, reusable stuff, but the few lines of code required to do so often beg the question of - "Why is this being configured?" 

The method name is usually a good way to answer that question - e.g. "configureAsyncProcessingForStreamingResponses". Each setting needs a comment as well - why 10 threads? Why 10 minutes? Why did we choose those numbers? 3 months from now, someone will look at this code and wonder why these settings were chosen.

I also think it's worth explicitly identifying the feature this is for - i.e. exporting entities. That may expand in the future - i.e. when we support exporting other results.

Reply:
Makes sense. I'll add the code comments.